File: Дорофеев/2101.cpp
class Solution {
public:
    double distance(double x1, double y1, double x2, double y2)
    {
        return sqrt(pow((x2 - x1), 2) + pow((y2 - y1), 2));
    }

    int DFS(int index, int x1, int y1, int r1,
        vector<vector<int>>& bombs, vector<int>& visited)
    {
        visited[index] = 1;
        int count = 1;
        for (int i = 0; i < bombs.size(); ++i)
        {
            if (visited[i]) continue;
            if (distance(x1, y1, bombs[i][0], bombs[i][1]) <= r1)
            {
                count += DFS(i, bombs[i][0], bombs[i][1], bombs[i][2], bombs, visited);
            }
        }

        return count;
    }

    int maximumDetonation(vector<vector<int>>& bombs)
    {
        int answer = 0;
        for (int i = 0; i < bombs.size(); ++i)
        {
            vector<int> visited(bombs.size(), 0);
            answer = max(answer, DFS(i, bombs[i][0], bombs[i][1], bombs[i][2], bombs, visited));
        }

        return answer;
    }
};

File: Volokitin/2101.py
class Solution:
    def maximumDetonation(self, bombs: List[List[int]]) -> int:
        graph = []
        amount = len(bombs)

        for i in range(amount):
            graph.append([])
            for j in range(amount):
                if i != j and length(bombs[i], bombs[j]) <= bombs[i][2]:
                    graph[i].append(j)

        max_bomb = 1

        for i in range(amount):
            visited = [0] * amount 
            curr_amount = dfs(graph, i, visited)  
            if curr_amount > max_bomb:
                max_bomb = curr_amount

        return max_bomb   
    
def dfs(graph, curr, visited):
    in_count = 1
    if visited[curr] == 0:
        visited[curr] = 1
        for i in graph[curr]:
            in_count += dfs(graph, i, visited)
        return in_count
    return 0

def length(first, second):
        return sqrt((first[0] - second[0]) ** 2 + (first[1] - second[1]) ** 2)